src/main/resources/static/index.html

```diff
@@ -60,23 +67,50 @@ <h2 class="card-pf-title">Display Report</h2>
     </div>
 </div>
 <script>
+var showNotification = function(msg) {
+  var html = '<div class="alert alert-warning alert-dismissable">' +
+    '<button type="button" class="close" data-dismiss="alert" aria-label="Close">' +
+    '<span class="pficon pficon-close"></span>' +
+    '</button><span class="pficon pficon-warning-triangle-o"></span>' + msg + '</div>';
+
+  $('#notifications').append(html);
+};
+
+var clearNotifications = function() {
+  $('#notifications').empty();
+};
+
+var clearTable = function() {
+  $('#table1').empty();
+};
+
 $('form').submit(false);
 var displayReport= function(){
-var userId= $("#userRating").val();
+  clearTable();
+  clearNotifications();
+  var userId= $("#userRating").val();
 
- $.get("/api/predictor/"+userId,
-        function(data, status){
-			var conf={ columns: data.columns,
-		            data: data.dataSet,
-		            dom: "t",
-		            order: [[ 0, 'asc' ]],
-		            pfConfig: {
-		              filterCaseInsensitive: true,
-		              paginationSelector: "#pagination1",
-		              colvisMenuSelector: '.table-view-pf-colvis-menu'
-		            },
-		         };
-		    var table= $("#table1").DataTable(conf);
+  $.ajax("/api/predictor/"+userId)
+    .done(function(data) {
+      if (data == "") {
+        showNotification("Warning: empty ratings response, please try again or check the predictor service");
+      }
+      else {
+        var conf={ columns: data.columns,
+          data: data.dataSet,
+          dom: "t",
+          order: [[ 0, 'asc' ]],
+          pfConfig: {
+            filterCaseInsensitive: true,
+            paginationSelector: "#pagination1",
+            colvisMenuSelector: '.table-view-pf-colvis-menu'
+          },
+        };
+        var table= $("#table1").DataTable(conf);
+      }
+    })
+    .fail(function() {
```

Comment:
so this is capturing any failure conditions caused by the query, separate from "query succeeded but returned empty result"  ?

Reply:
yes, after reviewing [the jquery docs](https://api.jquery.com/jQuery.get/), i noticed that we were using an older methodology for the `$.get`. modern javascript makes much more use of futures and those type of async patterns, so i figured just upgrade it.

the `.fail` clause is for any non-success status that is returned from the connection. so it should cover those cases we manually plucked out with the `status != "success"` stuff from before.